### PR TITLE
Add a class to facilitate creating read state engines for testing

### PR DESCRIPTION
**Pull request by akhaku**
This helper class is exposed in a new "hollow-test" jar.
This jar will contain utilities to aid in testing client
code that uses hollow.

### CODE REVIEW

File: hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private final HollowObjectMapper objectMapper;
+
+    private boolean built;
+
+    /**
+     * Create a HollowReadStateEngineBuilder with an empty initial type state. Adding objects will
+     * add their types, if not already present.
+     */
+    public HollowReadStateEngineBuilder() {
+        this(new ArrayList<Class<?>>());
```
Comment (davesu) [inline hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java]: Minor: Collectons.emptyList() ?

Comment (akhaku) [reply hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java]: Sure, will do. Note to self: make sure to specify the type ie `Collections.<Class<?>>.emptyList()`

File: hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        for (Class<?> type : types) {
+            objectMapper.initializeTypeState(type);
+        }
+    }
+
+    /**
+     * Add an object that will be in our built HollowReadStateEngine. You cannot add any more
+     * objects after calling build().
+     */
+    public HollowReadStateEngineBuilder add(Object object) {
```
Comment (davesu) [inline hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java]: I wonder if add(Object ... objs) will be good to have instead.

**Review by davesu — APPROVED** [review]

File: hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java (hunk 2 above)
Comment (akhaku) [reply hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java]: Can't hurt, that supports the current usage pattern and others, too. Will do.

**Review by davesu — APPROVED** [review]

**Pull request merged**